tensorflow_addons/layers/noisy_dense.py

```diff
@@ -0,0 +1,274 @@
+        self.kernel_constraint = constraints.get(kernel_constraint)
+        self.bias_constraint = constraints.get(bias_constraint)
+
+        self.input_spec = InputSpec(min_ndim=2)
+        self.supports_masking = True
+
+    def build(self, input_shape):
+        # Make sure dtype is correct
+        dtype = tf.dtypes.as_dtype(self.dtype or K.floatx())
+        if not (dtype.is_floating or dtype.is_complex):
+            raise TypeError(
+                "Unable to build `Dense` layer with non-floating point "
+                "dtype %s" % (dtype,)
+            )
+
+        input_shape = tf.TensorShape(input_shape)
+        self.last_dim = tf.compat.dimension_value(input_shape[-1])
+        sqrt_dim = self.last_dim ** (1 / 2)
+        if self.last_dim is None:
+            raise ValueError(
+                "The last dimension of the inputs to `Dense` "
+                "should be defined. Found `None`."
+            )
+        self.input_spec = InputSpec(min_ndim=2, axes={-1: self.last_dim})
+
+        self.σ_init = initializers.Constant(value=0.5 / sqrt_dim)
+        self.µ_init = initializers.RandomUniform(
+            minval=-1 / sqrt_dim, maxval=1 / sqrt_dim
+        )
+
+        # Learnable parameters
+        # Agent will learn to decay σ as it improves creating a sort of learned epsilon decay
+        self.σ_kernel = self.add_weight(
+            "σ_kernel",
+            shape=[self.last_dim, self.units],
+            initializer=self.σ_init,
+            regularizer=self.kernel_regularizer,
+            constraint=self.kernel_constraint,
+            dtype=self.dtype,
+            trainable=True,
+        )
+
+        self.µ_kernel = self.add_weight(
+            "µ_kernel",
+            shape=[self.last_dim, self.units],
+            initializer=self.µ_init,
+            regularizer=self.kernel_regularizer,
+            constraint=self.kernel_constraint,
+            dtype=self.dtype,
+            trainable=True,
+        )
+
+        if self.use_bias:
+            self.σ_bias = self.add_weight(
+                "σ_bias",
+                shape=[
+                    self.units,
+                ],
+                initializer=self.σ_init,
+                regularizer=self.bias_regularizer,
+                constraint=self.bias_constraint,
+                dtype=self.dtype,
+                trainable=True,
+            )
+
+            self.µ_bias = self.add_weight(
+                "µ_bias",
+                shape=[
+                    self.units,
+                ],
+                initializer=self.µ_init,
+                regularizer=self.bias_regularizer,
+                constraint=self.bias_constraint,
+                dtype=self.dtype,
+                trainable=True,
+            )
+        else:
+            self.σ_bias = None
+            self.µ_bias = None
+        self.built = True
+
+    @staticmethod
+    def _scale_noise(x):
+        return tf.sign(x) * tf.sqrt(tf.abs(x))
```

Comment:
I don't get the reason that it should be staticmethod. Maybe move it to the upper of class

```python
def _scale_noise(x):
    return tf.sign(x) * tf.sqrt(tf.abs(x))


class NoisyDense(xxx):
    xxx
```